app/code/Magento/InventoryLowQuantityNotification/Model/ResourceModel/BulkConfigurationTransfer.php

```diff
@@ -63,42 +63,60 @@ public function execute(
         $tableName = $this->resourceConnection->getTableName('inventory_low_stock_notification_configuration');
         $connection = $this->resourceConnection->getConnection();
 
-        $types = $this->getProductTypesBySkus->execute($skus);
+        $skuTypes = $this->getProductTypesBySkus->execute($skus);
 
-        foreach ($types as $sku => $type) {
-            if ($this->isSourceItemManagementAllowedForProductType->execute($type)) {
-                foreach ($skus as $sku) {
-                    $qry = $connection
-                        ->select()
-                        ->from($tableName, 'notify_stock_qty')
-                        ->where('sku = ?', $sku)
-                        ->where('source_code = ?', $originSource);
+        /**
+         * We are filtering SKU list for products which are not allowed to move configuration for
+         */
+        foreach ($skuTypes as $sku => $type) {
+            if (!$this->isSourceItemManagementAllowedForProductType->execute($type)) {
+                unset($skuTypes[$sku]);
+            }
+        }
+
+        $destinationItemsQuery = $connection
+            ->select()
+            ->from($tableName, ['sku', 'notify_stock_qty'])
+            ->where('source_code = ?', $destinationSource);
+        $destinationQueryResult = $connection->fetchPairs($destinationItemsQuery);
 
-                    $res = $connection->fetchOne($qry);
+        $destinationSkus = array_diff_key($skuTypes, $destinationQueryResult);
```

Comment:
looks like it's better to rename `destinationSkus ` variable as it represents not destination SKUs,

but SKUs to be migrated to destination Source